src/client/mapblock_mesh.cpp

```diff
@@ -319,9 +320,14 @@ void final_color_blend(video::SColor *result,
 	b += emphase_blue_when_dark[irr::core::clamp((s32) ((r + g + b) / 3 * 255),
 		0, 255) / 8] / 255.0f;
 
-	result->setRed(core::clamp((s32) (r * 255.0f), 0, 255));
-	result->setGreen(core::clamp((s32) (g * 255.0f), 0, 255));
-	result->setBlue(core::clamp((s32) (b * 255.0f), 0, 255));
+	// Add ambient light
```

Comment:
This does not seem to be gamma-correct to me, you're mixing sRGB values here. (That said, our existing code does not seem to be gamma-correct either...)